### PR TITLE
(#39) Implement basic game architecture

### LINKED ISSUE #39: Game basics
Before starting Basic gameplay tasks we need to implement basic game architecture in same way [Gyges](https://github.com/ForNeVeR/Gyges) project does.

### PR DESCRIPTION
**Pull request by gsomix**
Closes #39 

## Before merge [F]
- [x] TODO monitoring

### CODE REVIEW

File: O21.Game/Game.fs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        
+        spriteBatch <- new SpriteBatch(this.GraphicsDevice)
+        world <- game.Init()
+        base.Initialize()
+
+    override this.LoadContent() =
+        gameData <- game.LoadGameData (this.GraphicsDevice)
+        base.LoadContent()
+
+    override __.Update(gameTime) =
```
Comment (ForNeVeR) [inline O21.Game/Game.fs:70]:
```suggestion
    override _.Update(gameTime) =
```

**Review by ForNeVeR — APPROVED** [review]

**Comment by gsomix** [comment]
Thanks @sotiredofyours, I'm able to draw level now.

image

File: O21.Game/Game.fs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        
+        gd.SetRenderTarget(renderTarget)
+        spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend, SamplerState.PointClamp)
+        game.Draw spriteBatch gameData world
+        spriteBatch.End()
+
+        gd.SetRenderTarget(null)
+        spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.AlphaBlend, SamplerState.PointClamp)
+        spriteBatch.Draw(renderTarget, onScreenRect, Color.White)
+        spriteBatch.End()
```
Comment (ForNeVeR) [inline O21.Game/Game.fs:86]: Could you please explain this trick to me? Why do we call `SetRenderTarget` here twice?

File: O21.Game/Game.fs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        
+        base.Draw(gameTime)
+
+module GameState =
+    let create (config: Config) (game: Game<'World, 'GameData, 'Input>) =
+        let loop = new GameState<'World, 'GameData, 'Input>(config, game)
+        loop.IsFixedTimeStep <- config.IsFixedTimeStep
+        
+        let graphics = new GraphicsDeviceManager(loop)
```
Comment (ForNeVeR) [inline O21.Game/Game.fs:95]: This object is never disposed. Can we do better?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: O21.Game/Game.fs (hunk 2 above)
Comment (gsomix) [reply O21.Game/Game.fs:86]: We render in two steps:

- render everything on texture (setting render target) with dimensions same as original game screen
- render texture on actual window (render target not specified) with scaling (will be configurable)

File: O21.Game/Game.fs (hunk 3 above)
Comment (gsomix) [reply O21.Game/Game.fs:95]: [Should be disposed in Game class.](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Game.cs#L134)

File: O21.Game/Game.fs (hunk 2 above)
Comment (ForNeVeR) [reply O21.Game/Game.fs:86]: Ok, could you please add a comment about that into the code? This was the only really unobvious thing to me.

Comment (ForNeVeR) [reply O21.Game/Game.fs]: Ah well, I'll write that myself. Thanks!

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**